### PR TITLE
Adding greeting kata

### PR DESCRIPTION
**Pull request by melissamohan**

### CODE REVIEW

**Comment by melissamohan** [comment]
Fixes #13

File: greeting.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class Greeting
+  def greet(*name)
+    if name.any?
```
Comment (snasirca) [inline greeting.rb:3]: You can flip this `if` statement around by "returning early" which would reduce the nesting in the code. This is also known as a "guard clause": https://stackoverflow.com/a/37880756/998598

File: greeting.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        "Hello, #{name[0..-2].join(", ")}, and #{name.last}."
+      elsif name.length > 2 and name.include? up_case_string
+        "Hello, #{low_case[0..-2].join(', ')} and #{low_case.last}. AND HELLO #{up_case.join('')}!"
+      elsif name.length == 2 and name_string.include?(",") and  !name_string.include?("\"")
+        separated_names = split_names(name)
+        "Hello, #{separated_names[0..-2].join(', ')}, and#{separated_names.last}."
+      elsif name.length == 2 and name_string.include?("\"")
+        converted_names = escape_commas(name)
+        "Hello, #{converted_names.first} and #{converted_names[1..-2].join("")},#{converted_names.last}."
+      end
```
Comment (snasirca) [inline greeting.rb:24]: There is some duplication going on in the if statements which I think might actually be easier to read if you used nested if statements. So basically you'd have an outside 3 if statements (`name.length == 1`, `name.length == 2`, and `name.length > 2`. And then you'd have inner if statements that would check if it was all upper case/lower case, etc.

File: greeting.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      name_string = name.join("")
+      if name.length == 1 and name == low_case
+        "Hello, #{name.join(",")}."
+      elsif name.length == 1 and name == up_case
+        "HELLO #{name.join(",")}!"
+      elsif name.length == 2 and !name_string.include?(",") and !name_string.include?("\"")
+        "Hello, #{name.first} and #{name.last}."
+      elsif name.length > 2 and name == low_case
+        "Hello, #{name[0..-2].join(", ")}, and #{name.last}."
+      elsif name.length > 2 and name.include? up_case_string
```
Comment (snasirca) [inline greeting.rb:16]: Instead of using this `up_case_string`, why not just ask if `up_case.any?`? That is one less variable to know and reason about.

**Review by snasirca — APPROVED** [review]
Great work on this so far. I'd say merge it in so that you can close it off and work on improving it at your leisure. I've left some comments for improvements that you can make.

Notice that Requirement #6 is not fully met:

```
2.5.3 :007 > Greeting.new.greet("Jim", "BOB")
 => "Hello, Jim and BOB."
```

If I only provide 1 regular case name, it won't separate them into two sentences. This is why I think Its important to have 3 cases clearly laid out where you have only regular case names, upper case names, and a mixture of both.

File: greeting.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+class Greeting
+  def greet(*name)
+    if name.any?
+      low_case = name.reject { |x| x == x.upcase }
```
Comment (snasirca) [inline greeting.rb:4]: Its not really "lower case". Its "regular case".

**Comment by melissamohan** [comment]
@snasirca Awesome, thank you for your feedback. I will work on updating in the near future.

**Comment by melissamohan** [comment]
Fixes #13

**Pull request merged**